src/components/Reactions/Reaction.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+
+import {Button, ButtonSize, PaletteOption, PopoverProps, Popup} from '@gravity-ui/uikit';
+
+import {block} from '../utils/cn';
+
+import {useReactionsContext} from './context';
+import {useReactionsPopup} from './hooks';
+
+export type ReactionProps = Pick<PaletteOption, 'value' | 'content' | 'title'>;
+
+export interface ReactionStateProps {
+    /**
+     * Reaction's unique value (ID).
+     */
+    value: string;
+    /**
+     * Should be true when the user used this reaction.
+     */
+    selected?: boolean;
+    /**
+     * Display a number after the icon.
+     * Represents the number of users who used this reaction.
+     */
+    counter?: React.ReactNode;
+    /**
+     * If present, when a user hovers over the reaction, a popover appears with `tooltip.content`.
+     * Can be used to display users who used this reaction.
+     */
+    tooltip?: React.ReactNode;
+}
+
+interface ReactionInnerProps extends Pick<PaletteOption, 'content'> {
+    reaction: ReactionStateProps;
+    size: ButtonSize;
+    tooltipBehavior: 'tooltip' | 'popover';
```

Comment:
Do we really need that prop? Why it can't be default as popover?

Reply:
I'm not sure
In our services we always use the `popover` behavior
But someone might not need it. For example: someone can simply show users' avatars without any interactivity

Reply:
Removed the prop